src/geodesic.rs

```diff
@@ -614,7 +613,7 @@ impl Geodesic {
         const CARR_SIZE: usize = GEODESIC_ORDER as usize + 1;
         let mut C1a: [f64; CARR_SIZE] = [0.0; CARR_SIZE];
         let mut C2a: [f64; CARR_SIZE] = [0.0; CARR_SIZE];
-        let mut C3a: [f64; CARR_SIZE] = [0.0; CARR_SIZE];
+        let mut C3a: [f64; GEODESIC_ORDER as usize] = [0.0; GEODESIC_ORDER as usize];
```

Comment:
Looks like this small change more fully aligns us with the Python impl we were mostly, but not quite correctly, following.
```
    # real a12, sig12, calp1, salp1, calp2, salp2
    # index zero elements of these arrays are unused
    C1a = list(range(Geodesic.nC1_ + 1))
    C2a = list(range(Geodesic.nC2_ + 1))
    C3a = list(range(Geodesic.nC3_))
```

where:
```
  nC1_ = GEOGRAPHICLIB_GEODESIC_ORDER
  nC2_ = GEOGRAPHICLIB_GEODESIC_ORDER
  nC3_ = GEOGRAPHICLIB_GEODESIC_ORDER
```

Cpp seems to reuse a single `Ca` buffer for all its calculations rather than having three separate buffers. I'm not sure why the python implementation is so different from the C++ one... but agree that as long as we're following Python here, it makes sense to follow it more precisely. 👍